### PR TITLE
add bouncy build files

### PR DESCRIPTION
**Pull request by mikaelarguedas**
This doesnt add any source file yet. This will be done in a follow-up once we have dev job support on the ROS2 farm

### CODE REVIEW

File: bouncy/release-bionic-arm64-build.yaml
Comment (dirk-thomas) [inline bouncy/release-bionic-arm64-build.yaml:6]: The arm jobs shouldn't have the same priority as amd.

File: bouncy/release-bionic-arm64-build.yaml
Comment (dirk-thomas) [inline bouncy/release-bionic-arm64-build.yaml:12]: Please also create and use a google group (as in ROS 1).

Same below for amd as well as the index.

File: bouncy/release-bionic-arm64-build.yaml
Comment (mikaelarguedas) [reply bouncy/release-bionic-arm64-build.yaml:6]: They use the packet machines so it shouldn't be an issue.
I copied what we currently do for ardent. I can change them to have different priorities but we need to go back to modify previous distros (at least ardent that's still active) accordingly

File: bouncy/release-bionic-arm64-build.yaml
Comment (mikaelarguedas) [reply bouncy/release-bionic-arm64-build.yaml:12]: Fair point, I copied from ardent without looking further. I'll create a google group as we do for the ROS farm :+1:

File: bouncy/release-bionic-arm64-build.yaml
Comment (dirk-thomas) [reply bouncy/release-bionic-arm64-build.yaml:6]: Ok, if they are physically separate the priority doesn't make any difference. :+1:

File: bouncy/release-bionic-arm64-build.yaml
Comment (mikaelarguedas) [reply bouncy/release-bionic-arm64-build.yaml:12]: group has been created: https://groups.google.com/forum/#!forum/ros2-buildfarm-bouncy. Need to invite the buildfarm user once we have emails configured on the farm

**Comment by dirk-thomas** [comment]
> This will be done in a follow-up once we have dev job support on the ROS2 farm

With the branch from https://github.com/ros-infrastructure/ros_buildfarm/pull/548 devel jobs should use colcon and therefore be usable for Bouncy (at least we will see once the first jobs exist and run). Should I add a source build file to this PR or wait until this was merged and then create a separate PR for it?

**Comment by mikaelarguedas** [comment]
> Should I add a source build file to this PR or wait until this was merged and then create a separate PR for it?

Let's open a separate PR for it.
Once we have the other things sorted out like the smtp set up and first binary jobs working, we can enable dev/pr jobs.
Can you please create a separate PR targeting the ros2 branch ?
Thanks!

**Comment by dirk-thomas** [comment]
> Can you please create a separate PR targeting the ros2 branch ?

I will do that as soon as this PR has been merged (since that PR will need to register the new build file in the new Bouncy section).

**Review by wjwwood — APPROVED** [review]
lgtm, but again might want another's review.

**Review by nuclearsandwich — APPROVED** [review]

File: bouncy/release-bionic-arm64-build.yaml
Comment (dirk-thomas) [inline bouncy/release-bionic-arm64-build.yaml:14]: Is it intentional that these are disabled? Or when are they planned to be enabled if not now?

Comment (mikaelarguedas) [reply bouncy/release-bionic-arm64-build.yaml:14]: > Is it intentional that these are disabled?

Yes

>  Or when are they planned to be enabled if not now?

When we have email enabled on the ROS2 farm.

From https://github.com/ros2/ros_buildfarm_config/pull/5#issuecomment-393350851: `Once we have the other things sorted out like the smtp set up`

The task of configuring emails is tracked on the [release checklist](https://github.com/ros2/ros2/issues/493) under: "Infrastructure/configure build.ros2.org smtp"

Comment (dirk-thomas) [reply bouncy/release-bionic-arm64-build.yaml:14]: I don't see any harm in enabling this option now. The configuration already lists other emails for notification so Jenkins is going to try sending an email anyway.

Comment (mikaelarguedas) [reply bouncy/release-bionic-arm64-build.yaml:14]: Currently it is disabled in all build files of build.ros2.org.

As we didn't have email configured, I didn't see value in enabling it only in this file and open a follow-up to modify other (ardent) of incoming (source) files, neither it felt like this PR should touch unrelated files like the ardent ones.

If you think now is a good time, let's open a PR to change all files that should have this enabled :+1:

Comment (dirk-thomas) [reply bouncy/release-bionic-arm64-build.yaml:14]: > As we didn't have email configured, I didn't see value in enabling it only in this file

Well, I just saw it from the opposite side: there is no reason not to enable it - Jenkins is trying to send notifications anyway. So it just needs to enabled later. I will leave it up to you to enable it whenever you like.

Comment (dirk-thomas) [reply bouncy/release-bionic-arm64-build.yaml:14]: The email notification still needs to be enabled. The maintainers are currently not being notified about the outcome of the release builds.

Comment (mikaelarguedas) [reply bouncy/release-bionic-arm64-build.yaml:14]: Yes as discussed offline yesterday, maintainers should be updated to not spam unrelated maintainers and then we reenable emails. That is why https://github.com/ros2/ros2/issues/519 has been created

**Pull request merged**
